monai/losses/unified_focal_loss.py

```
@@ -213,10 +249,15 @@ def forward(self, y_pred: torch.Tensor, y_true: torch.Tensor) -> torch.Tensor:
-
-        if torch.max(y_true) != self.num_classes - 1:
+            if self.num_classes != 2:
+                raise ValueError(
+                    f"Single-channel input only supported for binary (num_classes=2), got {self.num_classes}"
+                )
+            y_pred = torch.cat([torch.zeros_like(y_pred), y_pred], dim=1)
+            if y_true.shape[1] == 1:
+                y_true = one_hot(y_true, num_classes=self.num_classes)
+
+        if y_true.shape[1] != self.num_classes and torch.max(y_true) > self.num_classes - 1:
             raise ValueError(f"Please make sure the number of classes is {self.num_classes-1}")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)